src/mpid/ch4/shm/ipc/src/ipc_mem.h

```diff
@@ -19,7 +20,16 @@ MPL_STATIC_INLINE_PREFIX int MPIDI_IPCI_get_mem_attr(const void *vaddr,
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_IPCI_GET_MEM_ATTR);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_IPCI_GET_MEM_ATTR);
 
-    mpi_errno = MPIDI_XPMEM_get_mem_attr(vaddr, attr);
+    MPL_pointer_attr_t mpl_attr;
+    MPL_gpu_query_pointer_attr(vaddr, &mpl_attr);
+
```

Comment:
Now I am a little concerned the cost to check the pointer attribute. `MPIDI_IPCI_get_mem_attr` is called at `MPIDI_IPC_mpi_isend` even for very small message. It definitely hurts intranode message rate. Maybe we want to define an IPC level P2P threshold. If the message is smaller than the threshold, we should directly fallback to POSIX. What do you think?

Reply:
FYI, falling back to POSIX will not avoid the cost of the check.  We'll need to check for it in POSIX too, for example to decide whether to do a copy or a DMA.  It depends on how much the cost of the check is, but we'll likely have a CVAR to turnoff GPU support anyway to avoid this cost for CPU-only applications.

Reply:
The CVAR for turning off GPU support is there.

Reply:
I see. If GPU support is on, do we want to pass down the pointer attribute to POSIX similar as that for AV? So POSIX does not have to do it again.